documents/coredocument.go

```diff
@@ -448,6 +465,56 @@ func (cd *CoreDocument) createProofs(fromZKTree bool, docType string, dataLeaves
 	}, nil
 }
 
+//CreateFingerprint generates a fingerprint for a Core Document
+func (cd *CoreDocument) CreateFingerprint() ([]byte, error) {
+	f := newFingerprint()
+	if len(cd.Document.Roles) == 0 {
```

Comment:
Looks like both Roles and TransitionRules have to be set to proceed. Maybe combine these two checks in one line?